api/src/OpenTelemetry/Trace/Core.hs

```diff
@@ -342,6 +348,13 @@ inSpan' ::
   m a
 inSpan' t = inSpan'' t callStack
 
+catchesHandler :: ExceptionEnricher -> SomeException -> [(Text, Attribute)]
+catchesHandler handler e = tryHandler handler
+  where 
+    tryHandler (ExceptionEnricher handler _)
+      = case fromException e of
+          Just e' -> handler e'
+          Nothing -> []
```

Comment:
hrm, I found the name shadowing a bit confusing here. Would probably prefer to inline some of that:

```suggestion
catchesHandler :: ExceptionEnricher -> SomeException -> [(Text, Attribute)]
catchesHandler (ExceptionEnricher handler _) e = 
  case fromException e of
    Just e' -> handler e'
    Nothing -> []
```